evaluation.md

```diff
@@ -0,0 +1,413 @@
+            addon_files = addon.get('current_version', {}).get('files', {})
+
+            is_webextension = any([f.get("is_webextension", False) for f in addon_files])
+            is_legacy = not is_webextension
+
+            if is_legacy:
+                legacy_addons.append(addon_id)
+            
+    return legacy_addons
+```
+
+## Completing client data
+
+
+```python
+def complete_client_data(client_data):
+    client = client_data.asDict()
+    
+    client['installed_addons'] = client['installed_addons'] or []
+    client['disabled_addon_ids'] = get_legacy_addons(client['installed_addons'])
+    client['locale'] = str(client['locale'])
+    
+    return client
+```
+
+## Evaluating the existing recommenders
+
+To check if a recommender is able to make a recommendation, it's sometimes easier and cleaner to directly query it instead of checking the important attributes ourselves. For example, this is the case for the locale recommender.
+
+
+```python
+from taar.recommenders import CollaborativeRecommender, LegacyRecommender, LocaleRecommender
+```
+
+
+```python
+class DummySimilarityRecommender:
+    def can_recommend(self, client_data):
+        REQUIRED_FIELDS = ["geoCity", "subsessionLength", "locale", "os", "bookmark_count", "tab_open_count",
+                           "total_uri", "unique_tlds"]
+
+        has_fields = all([client_data.get(f, None) is not None for f in REQUIRED_FIELDS])
+        return has_fields
+```
+
+
+```python
+recommenders = {
+    "collaborative": CollaborativeRecommender(),
+    "legacy": LegacyRecommender(),
+    "locale": LocaleRecommender(),
+    "similarity": DummySimilarityRecommender()
+}
+```
+
+
+```python
+def test_recommenders(client):
+    return tuple([recommender.can_recommend(client) for recommender in recommenders.values()])
+```
+
+## Computing combined counts
+
+We iterate over all clients in the longitudinal dataset, change the attributes to the expected format and then query the individual recommenders.
+
+
+```python
+from operator import add
+from collections import defaultdict
+```
+
+
+```python
+%%time
+results = rdd\
+    .map(complete_client_data)\
+    .map(test_recommenders)\
+    .map(lambda x: (x, 1))\
+    .reduceByKey(add)\
+    .collect()
+```
+
+    CPU times: user 9.76 s, sys: 632 ms, total: 10.4 s
+    Wall time: 25min 35s
+
+
+
+```python
+results = defaultdict(int, results)
+```
+
+
+```python
+num_clients = sum(results.values())
+```
+
+## Computing individual counts
+
+
+```python
+individual_counts = []
+
+for i in range(len(recommenders)):
+    count = 0
+    
+    for key, key_count in results.items():
+        if key[i]:
+            count += key_count
+            
+    individual_counts.append(count)
+```
+
+## Displaying the results
+
+
+```python
+from pandas import DataFrame
+```
+
+
+```python
+def format_int(num):
+    return "{:,}".format(num)
+```
+
+
+```python
+def format_frequency(frequency):
+    return "%.3f%%" % (frequency  * 100)
```

Comment:
don't multiply by 100 here, just show the raw numbers as they occured in your sample.